### DISCUSSION TITLE
The current PluginRemapper behavior is too intrusive

### DISCUSSION THREAD

**Discussion by wtlgo** [PaperMC/Paper#10649] [Ideas]
### Is your feature request related to a problem?

I noticed that paper 1.20.6 has something called "PluginRemapper" now. I don't quite understand yet what exactly it does, but I found it surprising that it casually overwrites the original .jar files in the plugins folder. None of the paper/spigot, forks has done it before, including paper and spigot itself. That's why I find this behavior not ideal: I don't expect the server software to modify other server's executable files in such an intrusive way.

### Describe the solution you'd like.

Instead of modifying the original .jar files, I would rather prefer the PluginRemapper to write the modified files somewhere else, e.g. `/cache/paper/remapped-plugins`, and load modified plugins from there.

### Describe alternatives you've considered.

As far as I understand, there's no way to alternate PluginRemapper's behavior to handle this issue at the moment.

### Other

Also, I would like to note that when Mohist did a similar thing in the past, the community deemed it as "malicious behavior". I'm not implying that this feature is malicious in any way, but still find it not ideal to say at least.

**Comment by electronicboy** [comment]
The plugin remaps a jar from spigots mappings over to papers (and redirects reflective calls to a tool intended to remap those); I wasn't aware that it modified jars in the plugins folder, that's news to me

(I'd also clarify that mohist wasn't remapping jars, they where downloading a jar off a 3rd party server to replace a jar you installed, entirely different scope of issues)

**Comment by wtlgo** [comment]
False alarm, It actually works in the way I described, instead, it writes to `plugins/.paper-remapped`

**Comment by electronicboy** [comment]
Okay, yea, I was correct in my original thoughts, paper does not modify the original plugin jars, they're remapped into a `.paper-remapped` inside of the plugins folder,; if you have stuff modifying your jars, I'd make sure that you don't have malware on your server
